src/Generator.php

```diff
@@ -153,6 +153,10 @@ public function createClassNameDetails(string $name, string $namespacePrefix, st
             }
         }
 
+        if (!str_ends_with($className, $suffix)) {
+            $className = sprintf('%s%s', $className, $suffix);
+        }
```

Comment:
Hmm... I don't think we should "fix" this here. One could argue that because you're passing an absolute class name into this method, it should not be altered. Under that assumption - this "fix" would actually be a BC break as it's changing the intended functionality from https://github.com/symfony/maker-bundle/commit/9e90cbc6d7d557187d3f097fdd0a7e1a883c7fd1#diff-4ee5eaad222ef0782fccd29473a5f1860988b7cf91cdcd0a12628b2377972b36

 Chewing on this for a better solution